astropy/units/core.py

```diff
@@ -1755,19 +1755,20 @@ def __call__(self, s, represents=None, format=None, namespace=None,
             if is_effectively_unity(represents.value):
                 represents = represents.unit
             else:
-                # cannot use _error_check=False: scale may be effectively unity
```

Comment:
This was actually not true: the `scale` was always checked (and with this PR that check is now very fast).